lib/src/socket_connector.dart

```diff
@@ -504,32 +526,40 @@ class SocketConnector {
     );
 
     StreamController<Socket> ssc = StreamController();
+    Mutex m = Mutex();
```

Comment:
Use a mutex to ensure that new connections to the bound ServerSocket are handled atomically, in strict sequence.